src/dygraph.js

```diff
@@ -3167,6 +3168,7 @@ Dygraph.prototype.updateOptions = function(input_attrs, block_redraw) {
 
   this.attributes_.reparseSeries();
 
+  if (prevNumAxes < this.attributes_.numAxes()) this.plotter_.clear();
```

Comment:
Would adding this condition to the `requiresNewPoints` check also fix the issue?

i.e.

```js
var requiresNewPoints = utils.isPixelChangingOptionList(this.attr_("labels"), attrs) ||
    prevNumAxes != this.attributes_.numAxes();
```

And does the check need to be `prevNumAxes != this.attributes_.numAxes()`?